SSO error handling

NobodysNightmare opened the pull request:
# Ticket
* https://community.openproject.org/projects/cross-application-user-integration-stream/work_packages/61612
* https://community.openproject.org/projects/cross-application-user-integration-stream/work_packages/61880

# What are you trying to accomplish?
Fixing errors related to showing the connection status of SSO users to their storage.

In the files tab, the following cases should now be indicated correctly:

* A "classic" (OAuth) user is not yet linked to the storage (existed before)
* A "classic" (OAuth) user could not successfully authenticate at the storage (new: indicated as an error)
* An SSO user could not successfully authenticate at the storage (indicated as an error)

In the project storages member overview (e.g. `/projects/:name/settings/project_storages/:id/members`):

* SSO members are now indicated as connected, once they obtained a remote identity
* SSO members have a specific not-connected message, when there is no remote identity yet

# Which approach did you choose and why?
It occured to me, that the question of which authentication method will really be used for a user at a storage has to be asked in many different places. Thus I extracted the relevant code into a helper called `AuthenticationMethodSelector`.

# Merge checklist

- [x] Added/updated tests
- [x] Tested major browsers (Chrome, Firefox, Edge, ...)

NobodysNightmare committed:
Differentiate error handling for storages

Introducing an explicit difference between users that were never linked
and those that have a link (i.e. a remote identity) but receive an authentication
error anyways. There is also differentiation between SSO users and non-SSO users.
When authenticating to the storage via SSO, it wouldn't make sense to show a login
button to users.

NobodysNightmare committed:
Fix display of member storage connection status

The display would not work at all for SSO-only storages
and would fail to recognize connections that didn't happen through
a storage's oauth client.

Now properly recognizing all remote identities and showing a special
warning message when a non-connected user would connect through SSO.

ba1ash commented on modules/storages/app/components/storages/project_storages/members/row_component.rb:
```diff
@@ -89,7 +86,12 @@ def principal_show_path
 
     def storage_connection_status
-      return :not_connected unless oauth_client_connected?
+      unless oauth_client_connected?
```
:yellow_circle: I think this method name is bad now. Therefore I suggest to remove the method and just use its body in the condition here. It is expressive enough and no need to think about a better method name.
```suggestion
      unless member.principal.remote_identities.exists?(integration: storage)
```

NobodysNightmare replied:
I changed the method name to `storage_connected?`. I preferred to keep the method, because it keeps the abstraction level inside `storage_connection_status` more consistent.

After renaming, I noticed that I could replace the `unless` with an `if` by switching the method order around, so I did that as well.

ba1ash replied:
Okay, works for me. Thanks for taking care! :+1:

ba1ash commented on modules/storages/app/components/storages/project_storages/members/row_component.rb:
```diff
@@ -99,12 +101,18 @@ def storage_connection_status
     end
 
     def oauth_client_connected?
```
:yellow_circle: I suggest to remove this method in favor the suggestion above.

ba1ash reviewed (approved):
LGTM. It works fine for my local setup. I have only one code suggestion.

NobodysNightmare committed:
Make recurring meeting specs work reliably

The way those specs were written they would not bbe able to
succeed before 10 AM in the morning, because meeting were scheduled
for 10:00, but the specs did not expect a meeting to be scheduled
today.

This fix is a workaround that makes sure we always run the specs
"after 10". A better fix would be to ALSO have specs running
before 10, and expecting the behaviour that's currently
seen as errorneous.
